Add transforms (#26)
holepunchto/mirror-drive: README.md

It starts processing all the diffing until is done.

## Transformers

> Apply content transformers during mirroring. Each item in `transformers` is a factory function `(key) => stream | null` that returns a new stream for that file. MirrorDrive always pipes through all provided transformers. If a transformer returns `null` (or a pass‑through), the file flows unchanged. Equality also runs through the transformers, so re‑runs emit `equal` when post‑transform bytes are unchanged.

Reviewer: ```suggestion
Apply content transformers during mirroring. Each item in `transformers` is a factory function `(key) => stream | null` that returns a new stream for that file. MirrorDrive always pipes through all provided transformers. If a `transformer` returns `null` it is safely skipped. Equality also runs through the transformers, so re‑runs emit `equal` when post‑transform bytes are unchanged.
```

Reviewer: > Equality also runs through the transformers, so re‑runs emit `equal` when post‑transform bytes are unchanged.

mirror-drive doesn't emit an equal event - can you rephrase, I know you mean the byte-diffs but it reads as if there's an equal event
